src/benchmark/operators/join_benchmark.cpp

```diff
@@ -9,103 +8,88 @@
-                                                 PredicateCondition::Equals);
-  warm_up->execute();
-  while (state.KeepRunning()) {
-    auto table_scan = std::make_shared<JoinSortMerge>(_table_wrapper_1, _table_wrapper_2, JoinMode::Inner,
-                                                      std::pair<ColumnID, ColumnID>(ColumnID{0}, ColumnID{0}),
-                                                      PredicateCondition::Equals);
-    table_scan->execute();
+  ColumnDataDistribution config = ColumnDataDistribution::make_uniform_config(0.0, 10000);
+  const auto chunk_size = static_cast<ChunkID>(number_of_rows / number_of_chunks);
+  Assert(chunk_size > 0, "The chunk size is 0 or less, can not generate such a table");
+
+  auto table = table_generator->generate_table(std::vector<ColumnDataDistribution>{config}, number_of_rows, chunk_size,
+                                               EncodingType::Dictionary, use_multiple_numa_nodes);
+
+  for (ChunkID chunk_id{0}; chunk_id < table->chunk_count(); ++chunk_id) {
+    auto chunk = table->get_chunk(chunk_id);
+
+    std::vector<ColumnID> columns{1};
+    for (ColumnID column_id{0}; column_id < chunk->column_count(); ++column_id) {
+      columns[0] = column_id;
+      chunk->create_index<AdaptiveRadixTreeIndex>(columns);
+    }
   }
+
+  auto table_wrapper = std::make_shared<TableWrapper>(table);
+  table_wrapper->execute();
+
+  return table_wrapper;
 }
 
-BENCHMARK_DEFINE_F(BenchmarkJoinFixture, BM_JoinMPSM)(benchmark::State& state) {
+template <class C>
+void BM_Join_impl(benchmark::State& state, std::shared_ptr<TableWrapper> table_wrapper_left,
+                  std::shared_ptr<TableWrapper> table_wrapper_right) {
   clear_cache();
 
   auto warm_up =
-      std::make_shared<JoinMPSM>(_table_wrapper_1, _table_wrapper_2, JoinMode::Inner,
-                                 std::pair<ColumnID, ColumnID>{ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals);
+      std::make_shared<C>(table_wrapper_left, table_wrapper_right, JoinMode::Inner,
+                          std::pair<ColumnID, ColumnID>{ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals);
   warm_up->execute();
   while (state.KeepRunning()) {
-    auto table_scan =
-        std::make_shared<JoinMPSM>(_table_wrapper_1, _table_wrapper_2, JoinMode::Inner,
-                                   std::pair<ColumnID, ColumnID>(ColumnID{0}, ColumnID{0}), PredicateCondition::Equals);
-    table_scan->execute();
+    auto join =
+        std::make_shared<C>(table_wrapper_left, table_wrapper_right, JoinMode::Inner,
+                            std::pair<ColumnID, ColumnID>(ColumnID{0}, ColumnID{0}), PredicateCondition::Equals);
+    join->execute();
   }
+
+  opossum::StorageManager::get().reset();
 }
 
-BENCHMARK_DEFINE_F(BenchmarkJoinFixture, BM_JoinNestedLoop)(benchmark::State& state) {
-  clear_cache();
+template <class C>
+void BM_Join(benchmark::State& state) {
+  auto table_wrapper_left = generate_table(100000, 50);
```

Comment:
What do you mean with kill the NLJ? The benchmarks run through, and yes, the NLJ ist slow compared to the others, but it does not crash for me.

Reply:
kill = running slow, yeah, that's what I meant. And if it would take, say, 6 minutes out of these 12 than it would make sense to drop the row_count as no one will ever use NLJ for 100k rows and we all know its slow. Just an idea. ;)